apm-agent-core/src/main/java/co/elastic/apm/agent/configuration/CoreConfiguration.java

```diff
@@ -129,12 +129,13 @@
         .tags("added[1.11.0]")
         .build();
 
-    private final ConfigurationOption<TimeDuration> delayInit = TimeDurationValueConverter.durationOption("ms")
```

Comment:
Is that on purpose that we have two distinct properties ? I'm not sure to get why we would need to split delay for tracer init and the premain execution.
- `delay_tracer_start` with alias `delay_tracer_start_ms` which is internal but documented in `CoreConfiguration`
- `delay_agent_premain_ms` property which does not rely on configuration to avoid any dependency.

Reply:
Yes, it is intentional, not saying it is the right thing 🙂 
Ideally, we wouldn't recommend anyone using the premain delay, unless we really have to. This is the absolute delay in all agent stuff, including all class loading and linkage. The disadvantages of using it are the need to reinstrument loaded classes (instead of applying instrumentation during class loading), which may have more side effects, as well as the more restricted configuration option (only as System property). We just had a customer lately getting strange JMX behaviour with specific WebLogic MBeans that was overcome through the delay in tracer start. For a large deployment, it might be less user-friendly to enforce changes in the startup scripts all across instead of applying the change through the agent config file.

I see your concern with having multiple properties dealing with delays, but since they are both hidden, it will mainly be our burden and not our users', so not as bad.

However, it got me thinking I will now break it for this user. Do you think we should:
-  just keep the old name?
-  use the new name and aliases for the old ones?
-  keep this change and notify this user, as it is internal and not likely widely used?